compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2461,6 +2461,24 @@ impl<'tcx> Ty<'tcx> {
             _ => None,
         }
     }
+
+    pub fn is_c_void(self, tcx: TyCtxt<'_>) -> bool {
+        match self.kind() {
+            ty::Adt(adt_def, ..) => {
+                let def_id = adt_def.did();
+                let crate_name = tcx.crate_name(def_id.krate);
+                if adt_def.is_enum()
+                    && tcx.item_name(def_id) == sym::c_void
+                    && crate_name == sym::core
```

Comment:
Could you make `c_void` a lang item? This is both faster and prevents confusion with a user crate named `core`.

Reply:
Do you have an example or could give me some directions on how to do that? I tried my best to do it for a type but couldn't figure it out.

Reply:
The list of lang items is around here: https://github.com/rust-lang/rust/blob/1a521db67e2935e5a46c7b95b511ab9a43be5770/compiler/rustc_hir/src/lang_items.rs#L248 Once you have added it you can add `#[cfg_attr(not(bootstrap), lang = "c_void")]` to the enum definition. You can then use code like in https://github.com/rust-lang/rust/blob/1a521db67e2935e5a46c7b95b511ab9a43be5770/src/tools/clippy/clippy_utils/src/ty.rs#L388 to check if this type is `c_void`.

Reply:
Thanks a lot for the help, @bjorn3! Done.